src/main/java/io/supertokens/webserver/api/bulkimport/DeleteBulkImportUserAPI.java

```
@@ -0,0 +1,114 @@
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.webserver.api.bulkimport;
+
+import java.io.IOException;
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+import io.supertokens.Main;
+import io.supertokens.bulkimport.BulkImport;
+import io.supertokens.multitenancy.exception.BadPermissionException;
+import io.supertokens.pluginInterface.Storage;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class DeleteBulkImportUserAPI extends WebserverAPI {
+    public DeleteBulkImportUserAPI(Main main) {
+        super(main, "");
+    }
+
+    @Override
+    public String getPath() {
+        return "/bulk-import/users/remove";
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        // API is app specific
+
+        if (StorageLayer.isInMemDb(main)) {
+            throw new ServletException(new BadRequestException("This API is not supported in the in-memory database."));
+        }
+
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+        JsonArray arr = InputParser.parseArrayOrThrowError(input, "ids", false);
+
+        if (arr.size() == 0) {
+            throw new ServletException(new WebserverAPI.BadRequestException("Field name 'ids' cannot be an empty array"));
+        }
```

in this case, just return status code 200 ok

---

in the right JSON output format ofc

---

I think its better to return an error because calling this API with no ids is not a valid case.

---

it makes it harder for people to call the API. Cause if they have a loop in which they are calling the API, they will be forced to check if the input is empty before calling the API (as opposed to being flexible enough to allow them to check it after and then exit the loop)

---

That makes sense. I have updated it.